proto/viam/app/mltraining/ml_training.proto

```diff
@@ -0,0 +1,68 @@
+syntax = "proto3";
+
+package viam.app.mltraining.v1;
+
+import "app/data/v1/data.proto";
+import "google/protobuf/struct.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "go.viam.com/api/app/mltrainining/v1";
+
+service MLTrainingService {
+  rpc SubmitTrainingJob(SubmitTrainingJobRequest) returns (SubmitTrainingJobResponse);
```

Comment:
As we add the UI, I think there are additional rpcs we're going to want here, such as GetTrainingJobs(orgid) GetRunningJobs(orgid) etc. I think we should wait until that's being worked on to add them so we actually know what we want/need, but calling that out